### PR TITLE
refactor: define errors in pkg

### PR DESCRIPTION
**Pull request by skoo87**
Define the special errors, so it's better to control the logic and
state.

Signed-off-by: skoo87 <marckywu@gmail.com>

**1.Describe what this PR did**
mgr is the core control layer, it should not contain the http state, because the http is a entry only and in the future, We will have the cri entry.  

**2.Does this pull request fix one issue?** 

**3.Describe how you did it**

**4.Describe how to verify it**

**5.Special notes for reviews**

### CODE REVIEW

**Comment by allencloud** [comment]
> mgr is the core control layer, it should not contain the http state 

I agree with this design. While I emphasize again that it needs us to construct our error message more strictly. We must wrap declared error like ErrNotFound, ErrInvalidParam,  when constructing returned error.

File: daemon/mgr/container.go [hunk 1]
```diff
@@ -190,7 +188,7 @@ func (mgr *ContainerManager) CreateExec(ctx context.Context, name string, config
 func (mgr *ContainerManager) StartExec(ctx context.Context, execid string, config *types.ExecStartConfig, attach *AttachConfig) error {
 	v, ok := mgr.ExecProcesses.Get(execid).Result()
 	if !ok {
-		return fmt.Errorf("exec process: %s not found", execid)
+		return errors.Wrap(ErrNotfound, "exec process: "+execid)
```
Comment (allencloud) [inline daemon/mgr/container.go]: The error message shows like `exec process: 12345678: not found`.
How about eliminate one `:` to be `exec process 12345678: not found`?

File: daemon/mgr/container.go [hunk 2]
```diff
@@ -247,7 +245,7 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 		}
 	}
 	if mgr.NameToID.Get(name).Exist() {
-		return nil, fmt.Errorf("container with name %s already exist", name)
+		return nil, errors.Wrap(ErrAlreadyExisted, fmt.Sprintf("container with name %s already exist", name))
```
Comment (allencloud) [inline daemon/mgr/container.go]: The error message shows like `container with name abcd already exist: already existed`.
How about simplifying to be `container abcd: already existed`?

File: apis/server/router.go [hunk 3]
```diff
@@ -121,7 +122,10 @@ func HandleErrorResponse(w http.ResponseWriter, err error) {
 	httpErr, ok := err.(httputils.HTTPError)
 	if ok {
 		code = httpErr.Code()
-		errMsg = httpErr.Error()
+	} else if errtypes.IsNotfound(err) {
+		code = http.StatusNotFound
+	} else if errtypes.IsInvalidParam(err) {
```
Comment (allencloud) [inline apis/server/router.go]: If we need to add code 409 here via codeAlreadyExisted?

**Comment by skoo87** [comment]
updated @allencloud

**Comment by allencloud** [comment]
LGTM

**Pull request merged**